### PR TITLE
[docs] Rename intake API

### PR DESCRIPTION
**Pull request by bmorelli25**
Closes https://github.com/elastic/apm-server/issues/3109.

* "Intake API" --> "API"
* "Events API" --> "Events intake API"
* Other cleanup as these files haven't been touched in a while.

### CODE REVIEW

File: docs/events-api.asciidoc [hunk 1]
```diff
@@ -90,20 +95,20 @@ An example error response might look something like this:
...
-* <<span-api, Spans>>
-* <<error-api, Errors>>
-* <<metricset-api, Metrics>>
-* <<example-intake-events, Example Request Body>>
+* <<metadata-api>>
+* <<transaction-api>>
+* <<span-api>>
+* <<error-api>>
+* <<metricset-api>>
+* <<example-intake-events>>
```
Comment (bmorelli25) [inline docs/events-api.asciidoc]: @simitt We don't have a section here for the new Jaeger endpoints. My gut instinct is that it isn't necessary, but I'm curious to hear your thoughts.

File: docs/events-api.asciidoc [hunk 2]
```diff
@@ -9,8 +13,9 @@ Events can be:
 * Spans
 * Errors
 * Metrics
+* Metadata
```
Comment (simitt) [inline docs/events-api.asciidoc]: Metadata need to be sent as first line of every request, but they are not an event category. The meta information is the same for every event of one request and stored in ES as part of each of these events. 
E.g. information about the monitored `service` or the installed `agent` is stored with every event but for efficiency reasons only sent once per request as `metadata`. 
In summary - please do not list it as event here.

File: docs/events-api.asciidoc (hunk 1 above)
Comment (simitt) [reply docs/events-api.asciidoc]: I am with you, the Jaeger data are sent via a different protocol and endpoint and is not part of the events intake API.

File: docs/intake-api.asciidoc [hunk 3]
```diff
@@ -21,16 +18,16 @@ The request body looks roughly like this:
 
 With NDJSON, agents can open an HTTP POST request and use chunked encoding to stream events to the APM Server as soon as they are recorded in the agent.
 This makes it simple for agents to serialize each event to a stream of newline delimited JSON.
-
 The APM Server also treats the HTTP body as a compressed stream and thus reads and handles each event independently.
```
Comment (simitt) [inline docs/intake-api.asciidoc]: All of the above description is part of the events Intake API and should be moved.

File: docs/events-api.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/events-api.asciidoc]: Thanks, will remove

**Pull request merged**